src/Program.cs

```diff
@@ -12,32 +13,50 @@ internal class Program
     {
         public static void Main(string[] args)
         {
-            var path = args.SingleOption().ToTry(_ => "Invalid arguments".ToEnumerable());
-            var archiveFiles = path.FlatMap(p => ZipFileReader.Read(p));
+            var (optionArguments, pathArguments) = args.Partition(a => a.StartsWith("--"));
+
+            var archivePath = pathArguments.SingleOption().ToTry(_ => "Invalid arguments".ToEnumerable());
+            var archiveFiles = archivePath.FlatMap(p => ZipFileReader.Read(p));
             var archive = archiveFiles.FlatMap(files => Archive.Create(files));
 
-            var result = archive.Match(
-                a => IsArchiveValid(a).Match(
+            var cryptoServiceProvider = GetCryptoServiceProvider(optionArguments);
+
+            var result = Try.Aggregate(
+                archive,
+                cryptoServiceProvider,
+                (a, rsa) => IsArchiveValid(a, rsa).Match(
                     t => "Archive signature IS valid.",
                     f => "Archive signature IS NOT valid."
-                ),
-                e => e.MkLines()
+                )
             );
-            Console.WriteLine(result);
+            Console.WriteLine(result.Match(r => r, e => e.MkLines()));
         }
 
-        private static bool IsArchiveValid(Archive archive)
+        private static ITry<RSACryptoServiceProvider, IEnumerable<string>> GetCryptoServiceProvider(IEnumerable<string> optionArguments)
         {
-            var computedSignature = ComputeSignature(archive);
-            var xmlKey = File.ReadAllText("PublicKey.xml");
-            var rsa = new RSACryptoServiceProvider();
-            rsa.FromXmlString(xmlKey);
+            var useDevelopKey = optionArguments.Contains("--develop", StringComparer.InvariantCultureIgnoreCase);
+            var fileName = useDevelopKey.Match(
+                t => "DevelopPublicKey.xml",
+                f => "ProductionPublicKey.xml"
+            );
+            var cryptoServiceProvider = Try.Create<RSACryptoServiceProvider, Exception>(_ =>
+            {
+                var xmlKey = File.ReadAllText(fileName);
+                var rsa = new RSACryptoServiceProvider();
+                rsa.FromXmlString(xmlKey);
+                return rsa;
+            });
+            return cryptoServiceProvider.MapError(_ => "Missing or invalid key.".ToEnumerable());
```

Comment:
Nothing wrong with some error handling. But is there an actual possibility of crashing other than the file not being found and/or the public key file not containing a valid key? 

Both of them are our problem and should be contract checked, so we might as well not try catch and display an error to user.

Reply:
👍 I initially considered this to be a user-problem due to the way we distribute the app. But if you delete another file from the distribution, it will throw an exception as well.